src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -386,17 +386,17 @@ TEST_F(EnsembleFlowCustomNodeAndDemultiplexerGatherPipelineExecutionTest, Multip
                                                                                              ovms::Precision::FP32,
                                                                                              ovms::Shape{1, 4},
                                                                                              Layout{"NC"})}};
-    nodes[0] = std::make_unique<EntryNode>(&predictRequest, inputsInfo);
+    nodes[0] = std::make_unique<EntryNode<PredictRequest>>(&predictRequest, inputsInfo);
     const tensor_map_t outputsInfo{{pipelineOutputName, dagDummyModelOutputTensorInfo}};
-    nodes[nodesCount - 1] = std::make_unique<ExitNode>(&response, outputsInfo);
+    nodes[nodesCount - 1] = std::make_unique<ExitNode<PredictResponse>>(&response, outputsInfo);
     size_t i = 1;
     for (size_t demultiplicationLayer = 0; demultiplicationLayer < demultiplicationLayersCount; ++demultiplicationLayer) {
         nodes[i++] = std::make_unique<CustomNode>(differentOpsNodeName + "-" + std::to_string(demultiplicationLayer), differentOpsLibrary, parameters_t{}, differentOpsOutputAlias, demultiplyCount);
         nodes[i++] = std::make_unique<DLNode>(dummyNodeName + "-" + std::to_string(demultiplicationLayer), "dummy", std::nullopt, modelManager);
         nodes[nodesCount - 1 - (i / 2)] = std::make_unique<CustomNode>(chooseMaxNodeName + "-" + std::to_string(demultiplicationLayer), chooseMaxLibrary, parameters, chooseMaxOutputAlias, std::nullopt, std::set<std::string>({differentOpsNodeName + "-" + std::to_string(demultiplicationLayer)}));
     }
 
-    Pipeline pipeline(dynamic_cast<EntryNode&>(*nodes[0]), dynamic_cast<ExitNode&>(*nodes[nodesCount - 1]));
+    Pipeline pipeline(dynamic_cast<EntryNode<PredictRequest>&>(*nodes[0]), dynamic_cast<ExitNode<PredictResponse>&>(*nodes[nodesCount - 1]));
```

Comment:
Why do you need to cast here? Pipeline accepts type "Node"

Reply:
Good catch